wire-ios-data-model/Tests/Source/Model/UserClient/UserClientTests.swift

```diff
@@ -1076,3 +1076,79 @@ extension UserClientTests {
     }
 
 }
+
+// MARK: - Proteus
+
+extension UserClientTests {
+
+    func test_GivenDeveloperFlagproteusViaCoreCryptoEnabled_itUsesCoreKrypto() {
+        // GIVEN
+        let context = self.syncMOC
+        var called = false
+        let prekey = "test".utf8Data!.base64String()
+        var resultOfMethod = false
+
+        let mockProteusService = MockProteusServiceInterface()
+        mockProteusService.establishSessionIdFromPrekey_MockMethod = {_, _ in
+            called = true
+        }
+        mockProteusService.remoteFingerprintForSession_MockMethod = {_ in
+            return "test"
+        }
+
+        let mock = MockProteusProvider(mockProteusService: mockProteusService, mockKeyStore: self.spyForTests())
+        mock.useProteusService = true
+
+        var sut: UserClient!
+        var clientB: UserClient!
+
+        context.performGroupedBlock {
+            sut = self.createSelfClient(onMOC: context)
+            let userB = self.createUser(in: context)
+            clientB = self.createClient(for: userB, createSessionWithSelfUser: false, onMOC: context)
+
+            // WHEN
+           resultOfMethod = sut.establishSessionWithClient(clientB, usingPreKey: prekey, proteusProviding: mock)
+
+            // THEN
+            XCTAssertTrue(called)
+            XCTAssertTrue(resultOfMethod)
+        }
+    }
+
+    func test_GivenDeveloperFlagproteusViaCoreCryptoDisabled_itUsesCryptoBox() {
+        // GIVEN
+        let context = self.syncMOC
+        var resultOfMethod = false
+        let prekey = "test".utf8Data!.base64String()
+        let spy = self.spyForTests()
+        let mock = MockProteusProvider(mockProteusService: MockProteusServiceInterface(), mockKeyStore: spy)
+        mock.useProteusService = false
+
+        var sut: UserClient!
+        var clientB: UserClient!
+
+        context.performGroupedBlock {
+            sut = self.createSelfClient(onMOC: context)
+            let userB = self.createUser(in: context)
+            clientB = self.createClient(for: userB, createSessionWithSelfUser: false, onMOC: context)
+            // we need real prekeys here
+            var preKeys: [(id: UInt16, prekey: String)] = []
+            sut.keysStore.encryptionContext.perform { sessionsDirectory in
+                preKeys = try! sessionsDirectory.generatePrekeys(0 ..< 2)
+            }
+
+            // WHEN
+            resultOfMethod = sut.establishSessionWithClient(clientB, usingPreKey: preKeys.first!.prekey, proteusProviding: mock)
+
+            // THEN
+            XCTAssertEqual(spy.accessEncryptionContextCount, 2)
```

Comment:
If you want some extra assertions, you could assert that the mock proteus service was never invoked.

Reply:
yes I'll add that